atd-etl/cr3_extract_diagram/cr3_extract_diagram_ocr_narrative.py

```diff
@@ -0,0 +1,374 @@
+      affected_rows
+        }
+    }
+    """
+    response = requests.post(
+        url=os.getenv("HASURA_ENDPOINT"),
+        headers={
+            "Accept": "*/*",
+            "content-type": "application/json",
+            "x-hasura-admin-secret": os.getenv("HASURA_ADMIN_KEY"),
+        },
+        json={
+            "query": query,
+            "variables": {"crash_id": crash_id, "metadata": metadata},
+        },
+    )
+    try:
+        return response.json()["data"]["update_atd_txdot_crashes"]["affected_rows"] > 0
+    except (KeyError, TypeError):
+        sys.stderr.write("ERROR")
+        sys.stderr.write(response.json())
+
+
+# Query for multiple unprocessed cr3s
+get_batch = """
+query find_cr3s($limit: Int) {
+  atd_txdot_crashes(where: {
+      cr3_ocr_extraction_date: {_is_null: true},
+      crash_id: {_gt: 10000}
+      crash_date: {_gte: "2015-01-01"}
+      cr3_file_metadata:{_is_null: false}
+  },
+  order_by: {crash_date: desc},
+  limit: $limit) {
+      crash_id,
+      cr3_ocr_extraction_date,
+      cr3_file_metadata
+  }
+}
+"""
+batch_variables = {"limit": int(args.batch_size)}
+
+# query for specific cr3, selected by crash_id
+get_single_cr3 = """
+query find_cr3s($crash_id: Int, $limit: Int) {
+  atd_txdot_crashes(where: {
+      crash_id: {_eq: $crash_id}
+      },
+    limit: $limit) {
+      crash_id,
+      cr3_ocr_extraction_date,
+      cr3_file_metadata
+  }
+}
+"""
+single_cr3_variables = {"crash_id": args.crash_id[0], "limit": int(args.batch_size)}
+
+graphql = ""
+variables = {}
+
+if args.crash_id[0]:
+    graphql = get_single_cr3
+    variables = single_cr3_variables
+else:
+    graphql = get_batch
+    variables = batch_variables
+
+response = requests.post(
+    url=os.getenv("HASURA_ENDPOINT"),
+    headers={
+        "Accept": "*/*",
+        "content-type": "application/json",
+        "x-hasura-admin-secret": os.getenv("HASURA_ADMIN_KEY"),
+    },
+    json={"query": graphql, "variables": variables},
+)
+
+for crash in response.json()["data"]["atd_txdot_crashes"]:
+    if args.v:
+        print("Preparing to operate on crash_id: " + str(crash["crash_id"]))
+
+    # build url and download the CR3
+    if args.v:
+        print("Pulling CR3 PDF from S3")
+    key = args.cr3_source[1] + "/" + str(crash["crash_id"]) + ".pdf"
+    obj = []
+    try:
+        pdf = s3.get_object(Bucket=args.cr3_source[0], Key=key)
+    except:
+        sys.stderr.write(f"Error: Failed to get PDF for crash {str(crash['crash_id'])}from the S3 object\n")
+        continue
+
+    # render the pdf into an array of raster images
+    if args.v:
+        print("Rendering PDF into images")
+    pages = []
+    try:
+        pages = convert_from_bytes(pdf["Body"].read(), 150)
+    except:
+        sys.stderr.write(
+            "Error: PDF Read for crash_id (" + str(crash["crash_id"]) + ") failed.\n"
+        )
+        continue
```

Comment:
I am getting an error when I run this from the docker container, 
```
[2023-07-12, 14:34:58 CDT] {docker.py:391} INFO - Preparing to operate on crash_id: 19638112
Pulling CR3 PDF from S3
[2023-07-12, 14:34:58 CDT] {docker.py:391} INFO - Rendering PDF into images
[2023-07-12, 14:34:59 CDT] {docker.py:391} INFO - Error: PDF Read for crash_id (19638112) failed.```

still trying to figure out why